variables.tf

```
@@ -46,6 +46,26 @@ variable "burstable_mode" {
   default     = null
 }
 
+variable "market_type" {
+  type        = string
+  description = "(Optional) Type of market for the instance. Valid values are `spot` and `capacity-block`. If the value set is `spot`, the `spot_options_attributes` variable becomes adjustable."
+  default     = null
+}
+
+variable "spot_options_attributes" {
+  type = list(object({
+    instance_interruption_behavior = string
+    max_price                      = number
+    spot_instance_type             = string
+    valid_until                    = string
+  }))
+  description = <<-EOT
+    Describes the market (purchasing) option for the instances.
+    To get more info see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#spot-options .
```

You can use markdown here to shorten the URL.

```suggestion
    Describes the market (purchasing) option for the instances.
    See [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#spot-options) for more information.
```